flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/GlobalBatchCommitterHandler.java

```diff
@@ -72,11 +58,12 @@ public void endInput() throws Exception {
             }
         }
         globalCommitter.endOfInput();
+        return Collections.emptyList();
     }
 
     @Override
     public void close() throws Exception {
-        super.close();
         globalCommitter.close();
+        super.close();
```

Comment:
Why do we need this change? I am not fully familiar with the exact semantics but first closing the object itself and next the `globalCommitter` looks safer.

Reply:
You are supposed to close resources always in the opposite order of creation. That's also how try-with-resource and Guava's Closer work. The main idea is that your later resources may depend on the prior resources to finish properly.
Here, it doesn't matter, but I'd still like to have it consistently correct in Flink.